src/gui/mzroll/tabledockwidget.cpp

```diff
@@ -1011,20 +1017,22 @@ void TableDockWidget::setGroupLabel(char label) {
       QVariant v = item->data(0, Qt::UserRole);
       PeakGroup *group = v.value<PeakGroup *>();
       if (group != NULL) {
-        if (!(group->label=='g'||group->label=='b')){
+        group->setLabel(label);
+        if (group->label=='g' || group->label=='b') {
```

Comment:
good/bad label was being checked before setting the label to ensure that changing the label for the same group (from 'g' to 'b' or vice versa) is not counted as a new group being marked.

Reply:
Corrected.